### PR TITLE
More clang-tidy fixes

### PR DESCRIPTION
**Pull request by cyyever**
cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: torch/csrc/distributed/c10d/FileStore.cpp [hunk 1]
```diff
@@ -313,7 +313,7 @@ FileStore::~FileStore() {
   // Clean up the file if number of references is 0.
   if (refCount == 0 && numWorkers_ >= 0 && numFinishedWorker >= numWorkers_) {
     // Best effort removal without checking the return
```
Comment (Skylion007) [inline torch/csrc/distributed/c10d/FileStore.cpp:315]: Why this one? std::prefix is better, no?

**Review by Skylion007 — APPROVED** [review]
This looks good except for the std::reduce nit. No reason to make the symbol more ambiguous.

File: torch/csrc/distributed/c10d/FileStore.cpp (hunk 1 above)
Comment (cyyever) [reply torch/csrc/distributed/c10d/FileStore.cpp:315]: std::remove is for STL containers to remove elements . ::remove is a C file API to remove a file

**Pull request closed without merging**